services/src/main/java/org/exoplatform/officeonline/documents/OfficeOnlineDocumentEditorPlugin.java

```diff
@@ -204,16 +308,33 @@ protected String getEditorLink(Node node, URI requestURI) {
    * @param node the node
    * @param requestURI the request URI
    * @return the string
+   * @throws RepositoryException the repository exception
+   * @throws OfficeOnlineException the office online exception
    */
-  protected String getEditorLink(Node node) {
-    String link = editorLinks.computeIfAbsent(node, n -> {
-      if (wopiService.canEdit(node)) {
-        return wopiService.getEditorLink(node, WOPIService.EDIT_ACTION);
-      } else if (wopiService.canView(node)) {
-        return wopiService.getEditorLink(node, WOPIService.VIEW_ACTION);
+  protected String getEditorLink(Node node, URI requestURI) throws RepositoryException, OfficeOnlineException {
+    String baseUrl;
+    if (requestURI != null) {
+      baseUrl = wopiService.platformUrl(requestURI.getScheme(), requestURI.getHost(), requestURI.getPort()).toString();
+    } else {
+      PortalRequestContext pcontext = Util.getPortalRequestContext();
+      if (pcontext != null) {
+        baseUrl = wopiService.platformUrl(pcontext.getRequest().getScheme(),
+                                          pcontext.getRequest().getServerName(),
+                                          pcontext.getRequest().getServerPort())
+                             .toString();
+      } else {
+        throw new OfficeOnlineException("Cannot get editor link - request URI and PortalRequestContext are null");
       }
-      return null;
-    });
+    }
+    String link = null;
+    if (wopiService.canEdit(node)) {
+      link = wopiService.getEditorLink(node, baseUrl, WOPIService.EDIT_ACTION);
```

Comment:
Let's pass all parts if the URL to the service without using platformUrl() here which should not be a public method.

Reply:
Fixed: https://github.com/exo-addons/office-online/pull/31/commits/ca043163690bc4c751e447a58539aceb6e4ce310